docs/SplunkOperatorUpgrade.md

```diff
@@ -0,0 +1,77 @@
+# How to upgrade Splunk Operator and Splunk Enterprise Deployments
+​
+To upgrade the Splunk Operator for Kubernetes, you will overwrite the prior operator release with the latest version. Once the latest version of the Splunk Operator is running, the CRD's are updated and any running pods in the operator managed namespace are restarted. Any new spec defined by the operator will be applied to the restarted pods.
```

Comment:
IMO this is the order in which we should do the update:

1. Update the CRD's
2. Update the operator deployment with a newer operator docker image version

Note: The **splunk-operator-install.yaml** also applies the CRDs before the deployment for the operator.

After step 1 above the K8S cluster, ETCD supports and stores both versions(old and new) versions of the CRD(i.e any CRs already deployed with either version) with the way we have structured our CRD's. But any new changes bought in with the new CRD will not be acted upon by the older version of the operator as the operator doesn't have the capability to understand and act upon the same(which is fine as older CRs with an older apiVersion for the CRD will still be served)

After step 2, the operator is well equipped to handle any such changes to the CRD.

If we reverse the order of the steps above, the Splunk Operator might be looking for some spec within a CR that has been deployed already and might not exist(the effects of which might not be catastrophic but rather unknown).

In general, having both versions of the CRDs stored in the database before the operator is upgraded might serve well for both existing deployments, future upgrades/deployments.

Would love to hear other thoughts if any.

Reply:
Do we want to make this distinction to the customers ? Considering we are offering customer to use splunk-operator-install.yaml which handles upgrade to both CRD and operator?

Reply:
I don't understand the comments here. It sounds like the upgrade order of ops is not understood.

Reply:
We can avoid the distinction completely or have the correct order.